docs/source/notebooks/quantum_kernel_methods.ipynb

```diff
@@ -0,0 +1,866 @@
+    "8 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; $ j \\leftarrow  j + 1$ <br>\n",
+    "9 &nbsp; &nbsp; &nbsp; **end while** <br>\n",
+    "10 &nbsp; &nbsp; &nbsp; $i \\leftarrow  i + 1$ <br>\n",
+    "11 &nbsp; **end while** <br>\n",
+    "12 &nbsp; $S \\leftarrow \\sqrt{K_Q} (K_C + \\lambda \\mathcal{I})^{-1} \\sqrt{K_Q}$ <br>\n",
+    "13 &nbsp; $g \\leftarrow$ max eigenvalue of $S$, $\\mathbf{v}_{g} \\leftarrow$ max eigenvector of $S$ <br>\n",
+    "14 &nbsp; $y \\leftarrow \\text{sign}(\\sqrt{K_Q} \\mathbf{\\mathbf{v}_{g}})$\n",
+    "***\n",
+    "\n",
+    "\n",
+    "This algorithm allows one to generate a dataset in which the geometric distance between the quantum and coherent kernels is maximised."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def generate_data(\n",
+    "    feature_map: Callable[[np.ndarray], Circuit], \n",
+    "    input_state: BasicState, \n",
+    "    data_size: int, \n",
+    "    data_dim: int, \n",
+    "    reg: float\n",
+    ") -> Tuple[np.ndarray, np.ndarray]:\n",
+    "    \"\"\"\n",
+    "    Generates ad hoc dataset for choice of feature map and initial state.\n",
+    "    Args: \n",
+    "        feature_map : Unitary to generate the dataset\n",
+    "        input_state : State on which the feature map acts on\n",
+    "        data_size   : Dataset size\n",
+    "        data_dim    : Feature dimension\n",
+    "        reg         : Regularization parameter\n",
+    "    Returns:\n",
+    "        Data features, Data labels\n",
+    "    \"\"\"\n",
+    "    # Generate features dataset\n",
+    "    X = np.random.uniform(0, 2, size=(data_size, data_dim))\n",
+    "\n",
+    "    # Construct quantum and coherent kernel matrices\n",
+    "    K_quantum = kernel_matrix(X, X, feature_map, input_state, indistinguishability=1)\n",
+    "    K_coherent = kernel_matrix(X, X, feature_map, input_state, indistinguishability=0)\n",
+    "    \n",
+    "    S = sqrtm(K_quantum) @ inv(K_coherent + reg * np.eye(data_size)) @ sqrtm(K_quantum)    \n",
+    "    eigenvals, eigenvecs = np.linalg.eig(S)\n",
+    "    v_g = eigenvecs[:, np.argmax(eigenvals)]\n",
+    "    vector = sqrtm(K_quantum) @ v_g\n",
+    "    \n",
+    "    # Assign labels\n",
+    "    y = np.where(np.real(vector) >= 0, 1, -1)\n",
+    "    return X, y"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will generate a feature map which encodes each component of a feature vector into the phase shifters of our linear optical set-up. Specifically, we let each pair of components be encoded into the phase shifters of a Mach-Zender interferometer (MZI): $\\phi_i \\rightarrow 2 \\pi x_i$.\n",
+    "\n",
+    "Each MZI is arranged in a brickwork pattern as shown in the figure below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
```

Comment:
Can you clean the outputs ?
Notebooks are executed anyway when making a release and cleaning makes the repo lighter

Reply:
I guess all other notebooks contain their outputs (even large pictures and graphs). For a newcomer (such as me on this notebook), it's easier to check what's up when the outputs are already generated.
Plus, I believe the rerun is not used to generate the doc, only to check notebook syntax. I'd rather have the outputs stay.